### PR TITLE
chore(prisma): undo skipped type checks

### PR DESCRIPTION
**Pull request by millsp**
## What does this PR do?

This PR restores the types as they mostly were before the Prisma Accelerate extension was used. The original issue that prompted introducing a bunch of `@ts-expect-error` is tracked in the Prisma repository https://github.com/prisma/prisma/issues/21136.

## Requirement/Documentation

n/a

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

n/a

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

✅

### CODE REVIEW

**Review by zomars — COMMENTED** [review]
Code changes are looking good so far 🙏🏽

File: apps/web/test/utils/bookingScenario.ts [hunk 1]
```diff
@@ -245,16 +242,13 @@ function addUsers(users: InputUser[]) {
   // @ts-ignore
```
Comment (millsp) [inline apps/web/test/utils/bookingScenario.ts:242]: Should these also be removed?

Comment (zomars) [reply apps/web/test/utils/bookingScenario.ts:242]: Let's try that for sure

File: apps/web/test/lib/handleChildrenEventTypes.test.ts [hunk 2]
```diff
@@ -40,8 +39,7 @@ describe("handleChildrenEventTypes", () => {
         currentUserId: 1,
         hashedLink: undefined,
         connectedLink: null,
-        // @ts-expect-error Prisma v5 typings are not yet available
-        prisma: prismaMock,
+        prisma: prismaMock as PrismaClient,
```
Comment (zomars) [inline apps/web/test/lib/handleChildrenEventTypes.test.ts:42]: Can we cast directly on the prismaMock source? So we don't have to repeat this everywhere?

**Comment by millsp** [comment]
I also noticed that your mocking library could be slowing down your VSCode experience. I opened a PR https://github.com/eratio08/vitest-mock-extended/pull/282

File: packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts [hunk 3]
```diff
@@ -41,7 +42,7 @@ interface handleChildrenEventTypesProps {
         };
       }[]
     | undefined;
-  prisma: PrismaClient;
+  prisma: PrismaClient | DeepMockProxy<PrismaClient>;
```
Comment (zomars) [inline packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:45]: Why is it only needed here? Is it really needed tho?

**Comment by zomars** [comment]
Thank you for your contribution 🙏 Will merge this into my upgrade branch to follow up.

**Pull request merged**